pkg/authorization/authz.go

```diff
@@ -76,6 +77,13 @@ func (ctx *Ctx) AuthZRequest(w http.ResponseWriter, r *http.Request) error {
 		RequestHeaders:  headers(r.Header),
 	}
 
+	if r.TLS != nil {
+		for _, c := range r.TLS.PeerCertificates {
```

Comment:
Typically, we do something like this by declaring `Marshal`/`Unmarshal` on a special type. This will avoid bugs under refactor.